woocommerce/woocommerce
# Add a loading mask to the order summary when pushing data to the server

### Changes proposed in this Pull Request:

- Introduces a new property in the `wc/store/cart` data store (`applyingExtensionCartUpdates`) this is used to track the current number of calls to `applyExtensionCartUpdate`. Using this, we can determine if some server-side work that might update the order summary is currently happening.
- Apply a `LoadingMask` to the order summary block if a push is happening, e.g. selecting a new shipping rate, updating the customer address, or `applyExtensionCartUpdate`.
- Disable the place order button if any `applyExtensionCartUpdate`s are being executed.



<img width="562" alt="image" src="https://github.com/user-attachments/assets/f828954b-d786-409f-92bb-6159c0f4ff84">


Closes #51565.



### How to test the changes in this Pull Request:

1. Install and activate this test plugin: [test-plugin.zip](https://github.com/user-attachments/files/17924414/test-plugin.zip)
2. Set up at least 2 shipping rates on your store. You'll need to switch between them later.
3. Set up Local Pickup (WooCommerce -> Settings -> Shipping -> Local Pickup) and add at least 2 pickup locations.
4. Add an item to your cart and go to the Checkout block.
5. Notice there's a control that lets you simulate a `cartExtensionUpdate` - this is from the test plugin.
<img width="538" alt="image" src="https://github.com/user-attachments/assets/1e473510-c6bf-4fbf-9cb1-00b87e520208">

6. Press this button, expect the order summary to fade out and show a spinner for approx the same number of seconds in the input. Try with a few different values, 1, and 10 for example.
7. Change your address, ensure the address is "complete" (e.g. it has a country, state, address 1, and postcode).
8. Expect the order summary to fade out and show a spinner.
9. Change the shipping method. Expect the order summary to fade out and show a spinner.
10. Switch to local pickup. Change the pickup method. Expect the order summary to fade out and show a spinner.
11. Go back to the `Delay extension response for 5 seconds` input. Press it with a high number.
12. Scroll down to the place order button and try clicking it, it should be faded out, and it should not place the other while the delayed `extensionCartUpdate` is processing.
13. When it is finished processing, ensure you can place the order successfully.

See the testing video below

https://www.loom.com/share/a4783b72adb04dca9e5d1460994c8315



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
